spec.html

```diff
@@ -1097,27 +1103,46 @@ <h1>DecorateClass ( elements, decorators )</h1>
 
     <emu-clause id=sec-decorate-element aoid=DecorateElement>
       <h1>DecorateElement ( element, keys )</h1>
-      <p>With parameters _element_, a Class Element and _keys_, a List of Property Keys and Private Names.</p>
+      <p>With parameters Class Element _element_ and _keys_:</p>
       <emu-alg>
         1. Let _extras_ be a new empty List.
         1. Let _finishers_ be a new empty List.
+        1. Let _placementKeys_ be *undefined*.
         1. For each _decorator_ in element_.[[Decorators]], in reverse list order do
-          1. Assert: _element_.[[Key]] is an element of _keys_.
-          1. Remove _element_.[[Key]] from _keys_.
+          1. If _element_.[[Placement]] is `"own"`, then
+            1. set _placementKeys_ to _keys_.[[OwnKeys]].
+          1. Else if _element_.[[Placement]] is `"static"`, then
+            1. set _placementKeys_ to _keys_.[[StaticKeys]].
+          1. Else,
+            1. set _placementKeys_ to _keys_.[[PrototypeKeys]].
```

Comment:
Since this if/else if/else steps are used different times, maybe we can extract them into a different "function"?
e.g.
```
1. Let _placementKeys_ be GetPlacementKeys(_keys_, _element_.[[Placement]])
```

Reply:
I didn't want to pollute the spec with an additional tangential function.

Reply:
Personally I’d prefer *more* abstract operations if it can clean up the code, but generally it’s best to only do so when things are very complex or needed in more than one place.